src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -82,13 +80,9 @@ class AgoraBusiness {
                  entityTypes: Seq[AgoraEntityType.EntityType],
                  username: String): AgoraEntity = {
     val foundEntity = AgoraDao.createAgoraDao(entityTypes).findSingle(namespace, name, snapshotId)
-
-    if (!AgoraEntityPermissionsClient.getEntityPermission(foundEntity, username).canRead &&
-        !AgoraEntityPermissionsClient.getEntityPermission(foundEntity, "public").canRead) {
-      throw new AgoraEntityNotFoundException(foundEntity)
-    }
-
-    foundEntity.addUrl().removeIds()
+    val entities = AgoraEntityPermissionsClient.filterEntityByRead(Seq(foundEntity), username)
+    if (entities.size == 0) throw new AgoraEntityNotFoundException(foundEntity)
+    else entities.head.addUrl().removeIds().addManagers(AgoraEntityPermissionsClient.listOwners(foundEntity))
```

Comment:
I'd want to check the weird corner case where size > 1 here. Your current code won't throw an error on this, and I think we should, because it's totally unexpected. You could check `entities.size == 1`, or you could get all Scala-happy and pattern-match it, looking for a sequence of one element:
```
    entities match {
      case Seq(ae:AgoraEntity) => ae.addUrl().removeIds().addManagers(AgoraEntityPermissionsClient.listOwners(foundEntity))
      case _ => throw new AgoraEntityNotFoundException(foundEntity)
    }
```